src/test/regress/expected/with_modifying.out

```diff
@@ -822,6 +822,33 @@ EXECUTE olu(1,ARRAY[1,2],ARRAY[1,2]);
  {1}       | {1}       | {NULL}
 (1 row)
 
+-- test insert query with insert CTE
+WITH insert_cte AS
+       (INSERT INTO with_modifying.modify_table VALUES (23, 7))
+INSERT INTO with_modifying.anchor_table VALUES (1998);
+SELECT * FROM with_modifying.modify_table WHERE id = 23 AND val = 7;
+ id | val
+---------------------------------------------------------------------
+ 23 |   7
+(1 row)
+
+SELECT * FROM with_modifying.anchor_table WHERE id = 1998;
+  id
+---------------------------------------------------------------------
+ 1998
+(1 row)
+
+-- test insert query with select CTE
+WITH select_cte AS
```

Comment:
`select_cte ` is not used at all, and both Citus/PG would ignore this CTE anyway. Can we write a query where we use the output of the CTE?

Reply:
probably the only thing that works is using another modifying CTE that uses the select_cte, since sublinks in the insert are not supported

Reply:
It's not ignored when the query is an insert query with a select CTE. If it uses router plan, query after rebuilding is not correct.

```SQL
create table table1 (a int);
create table table2 (a text);
select create_distributed_table('table1','a');
select create_distributed_table('table2','a');
with wcte as (select * from table1)
  insert into table2 values ('hello world');
```
Errors out as:
```SQL
DEBUG:  StartTransaction(1) name: unnamed; blockState: DEFAULT; state: INPROGRESS, xid/subid/cid: 0/1/0
DEBUG:  Creating router plan
DEBUG:  query has a single distribution column value: hello world
DEBUG:  query before rebuilding: (null)
DEBUG:  query after rebuilding:  WITH wcte AS (SELECT table1.a FROM public.table1) INSERT INTO public.table2_102063 (a) VALUES ('hello world'::text)
DEBUG:  assigned task 0 to node localhost:9702
DEBUG:  opening 1 new connections to localhost:9702
DEBUG:  established connection to localhost:9702 for session 10
NOTICE:  issuing WITH wcte AS (SELECT table1.a FROM public.table1) INSERT INTO public.table2_102063 (a) VALUES ('hello world'::text)
DETAIL:  on server postgres@localhost:9702 connectionId: 7
ERROR:  relation "public.table1" does not exist
CONTEXT:  while executing command on localhost:9702
```

This is why I checked this case at `multi_router_planner.c`, and then added a test for it to make sure it doesn't error out.
@onurctirtir also has more detail for your question https://github.com/citusdata/citus/pull/4700#discussion_r578303982

I can try adding a test like @marcocitus  suggested.